cluster-autoscaler/config/migconfig.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type MigConfig struct {
+	MinSize int
+	MaxSize int
+	Url     url.URL
+}
+
+type MigConfigFlag []MigConfig
+
+func (this *MigConfigFlag) String() string {
+	var b bytes.Buffer
+	b.WriteString("[")
+	for i, npc := range *this {
+		if i > 0 {
+			b.WriteString(" ")
+		}
+		b.WriteString(fmt.Sprintf("%d:%d:%s", npc.MinSize, npc.MaxSize, npc.Url.String()))
+	}
+	b.WriteString("]")
+	return b.String()
+}
+
+func (this *MigConfigFlag) Set(value string) error {
+	tokens := strings.SplitN(value, ":", 3)
+	if len(tokens) != 3 {
+		return fmt.Errorf("wrong nodes configuration: %s", value)
+	}
+	ncf := MigConfig{}
+	if size, err := strconv.Atoi(tokens[0]); err == nil {
+		ncf.MinSize = size
+	} else {
+		return fmt.Errorf("failed to set min size: %s", tokens[0])
+	}
+
+	if size, err := strconv.Atoi(tokens[1]); err == nil {
+		ncf.MaxSize = size
+	} else {
+		return fmt.Errorf("failed to set max size: %s", tokens[1])
+	}
+
+	if url, err := url.Parse(tokens[2]); err == nil {
+		ncf.Url = *url
+	} else {
+		return fmt.Errorf("failed to parse mig url: %s", tokens[2])
+	}
+	if len(ncf.Url.String()) < 5 {
```

Comment:
url.Parse accepts empty strings and strings that do not look like urls. anything less than couple charecters is most likely not a valid mig.